File: Python/Domotica/LampadinaDimmerabile.py
from LampadinaSmart import LampadinaSmart
class LampadinaDimmerabile(LampadinaSmart):
    def __init__(self, stato=False, consumo=0):
        super().__init__(stato, consumo)
    def accendi(self, intensita):
        super().accendi()
        self.intensita = intensita
        print(f"Lampadina accesa con intensità {self.intensita}")

File: Java/domoticaJava/src/LampadinaDimmerabile.java
public class LampadinaDimmerabile extends LampadinaSmart {
    private int intensita;

    public LampadinaDimmerabile(double consumo) {
        super(consumo);
    }
    public void accendi(int intensita) {
        super.accendi();
        this.intensita = intensita;
        System.out.println("Lampadina dimmerabile accesa con intensita: " + intensita);
    }
}
